### PR TITLE
feat: add create and edit project screen (NEW)

### PR DESCRIPTION
**Pull request by ykhedher**

### CODE REVIEW

**Comment by ykhedher** [comment]
[Create screen - Project](https://trello.com/c/4LK8V8vH/684-create-screen-project)

File: src/component/project/Project/hooks/useProjectForm.ts [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+    const getProjectPayload = () => {
+        return {
+            id: projectId,
+            name: projectName,
+            description: projectDesc,
+        };
+    };
+
+    const validateIdUniqueness = () => {
```
Comment (FredrikOseberg) [inline src/component/project/Project/hooks/useProjectForm.ts:40]: This is not good enough to validate the id. We need to call an endpoint to do this on the server side. The existing project performs the validation like this: 

```
    const validateId = async (id: string) => {
        if (editMode) return true;

        const e = { ...errors };
        try {
            await props.validateId(id);
            e.id = undefined;
        } catch (err: any) {
            e.id = err.message;
        }

        setErrors(e);
        if (e.id) return false;
        return true;
    };

```

File: src/component/project/Project/ProjectForm/ProjectForm.tsx [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+                />
+            </div>
+
+            <div className={styles.buttonContainer}>
+                <Button onClick={handleCancel} className={styles.cancelButton}>
+                    Cancel
+                </Button>
+                <PermissionButton
+                    onClick={handleSubmit}
+                    permission={ADMIN}
```
Comment (FredrikOseberg) [inline src/component/project/Project/ProjectForm/ProjectForm.tsx:92]: Only admins can create project? This should be CREATE_PROJECT permission.

File: src/component/project/Project/ProjectForm/ProjectForm.tsx [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+                    type="submit"
+                >
+                    {submitButtonText} project
```
Comment (FredrikOseberg) [inline src/component/project/Project/ProjectForm/ProjectForm.tsx:95]: This should no longer be needed. We have separate components now for Create and Edit.

File: src/component/project/Project/ProjectForm/ProjectForm.tsx [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+    return (
+        <form onSubmit={handleSubmit} className={styles.form}>
+            <h3 className={styles.formHeader}>Project Information</h3>
+
+            <div className={styles.container}>
+                <p className={styles.inputDescription}>
+                    What is your project Id?
+                </p>
+                <Input
+                    className={styles.input}
```
Comment (FredrikOseberg) [inline src/component/project/Project/ProjectForm/ProjectForm.tsx:48]: We need to make sure this field does not allow spaces. Currently we are doing it like this in the old view: 

```
    <TextField
          label="Project Id"
          name="id"
          placeholder="A-unique-key"
          value={project.id}
          error={!!errors.id}
          helperText={errors.id}
          disabled={editMode}
          variant="outlined"
          size="small"
          onBlur={v => validateId(v.target.value)}
          onChange={v =>
              setValue('id', trim(v.target.value))
          }
      />
```

**Review by FredrikOseberg — CHANGES_REQUESTED** [review]
Needs a few changes before it's ready.

**Pull request merged**